File: AutomaticDrawing.Recognition.Clustering/Recognition/ClusteringRecognition.vb
Imports AutoPaint.Core
''' <summary>
''' 使用聚类模型的线条识别器
''' </summary>
Public Class ClusteringRecognition
    Implements IRecognition

    Private Function IRecognition_Recognize(pixels As PixelData) As IPainting Implements IRecognition.Recognize
        Dim clusterAI As New ClusterAI(pixels)
        Return New Painting() With {.Lines = clusterAI.Lines}
    End Function
End Class

File: AutoPaint.Recognition.Clustering/Recognition/ClusteringRecognition.cs
using AutoPaint.Core;
using System;
using System.Collections.Generic;
using System.Text;

namespace AutoPaint.Recognition.Clustering.Recognition
{
    public class ClusteringRecognition : IRecognition
    {
        public IPainting Recognize(PixelData pixels)
        {
            ClusteringAI clusteringAI = new ClusteringAI(pixels);
            return new Painting() { Lines = clusteringAI.Lines };
        }
    }
}
